tests/test_database.py

```diff
@@ -0,0 +1,271 @@
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# pylint: disable=useless-suppression,missing-function-docstring,missing-class-docstring
+# pylint: disable=too-many-statements,too-many-instance-attributes,missing-return-doc
+# pylint: disable=missing-return-type-doc,no-value-for-parameter,protected-access,
+# pylint: disable=too-many-public-methods,no-self-use
+
+import unittest
+from unittest import mock
+
+from pathlib import Path
+
+from astarte.device import database
+
+TEST_DATABASE = Path("./test_database_DELETE_ME.db")
+
+
+class UnitTests(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    @mock.patch("astarte.device.database.sqlite3.connect")
+    def test_initialize(self, mock_sqlite3_connect):
+        path_to_database = mock.MagicMock()
+
+        database.AstarteDatabaseSQLite(path_to_database)
+
+        mock_sqlite3_connect.assert_called_once_with(path_to_database)
+        mock_sqlite3_connect.return_value.cursor.assert_called_once_with()
+        execute_expected_arg = "CREATE TABLE IF NOT EXISTS properties (interface TEXT, major INTEGER NOT NULL, path TEXT, value BLOB NOT NULL, PRIMARY KEY (interface, path))"
```

Comment:
Same as above, maybe it could be a constant for the crate query shared between the two modules

Reply:
Do you mean if it could be a constant for the class shared between the two methods?